src/sisl/io/wannier90/seedname.py

```diff
@@ -297,69 +278,133 @@ def _r_wigner_seitz_weights(self):
         ws = 1.0 / _a.arrayd(ws)
         return ws
 
+
+    def _r_hamiltonian_tb(self, *args, **kwargs):
+        """Read Hamiltonian from the <>_tb.dat file"""
+        f = self.dir_file(self._seed + "_tb.dat")
+        H = None
+        if f.exists():
+            H = tbSileWannier90(f).read_hamiltonian(*args, **kwargs)
+        return H
+
+    def _r_hamiltonian_hr(self, *args, **kwargs):
+        """Reads a Hamiltonian model from the <>_hr.dat file"""
+        f = self.dir_file(self._seed + "_hr.dat")
+        H = None
+        if f.exists():
+            H = hrSileWannier90(f).read_hamiltonian(*args, **kwargs)
+        return H
+
     @sile_fh_open()
-    def _r_hamiltonian_hr(self, geometry, dtype=np.float64, **kwargs):
-        """Reads a Hamiltonian model from the _hr.dat file"""
-        cutoff = kwargs.get("cutoff", 0.00001)
+    def read_hamiltonian(self, cutoff: float = 1e-5, *args, **kwargs):
+        """Read the electronic structure of the Wannier90 output by reading the <>_tb.dat, <>_hr.dat
+
+        Parameters
+        ----------
+        cutoff:
+           the cutoff value for the zero Hamiltonian elements, default
+           to 0.00001 eV.
+
+        dtype: np.float64, optional
+            the default data-type used for the matrix.
+            Is mainly useful to check whether the TB model has imaginary
+            components (it should not since it is a Wannier model).
+        """
+        order = kwargs.pop("order", ["tb", "hr"])
+
+        if "geometry" not in kwargs:
+            # to ensure we get the correct orbital positions
+            kwargs["geometry"] = self.read_geometry()
```

Comment:
perhaps be explicit here, so it is clear in the code what we are doing? `order=centres`